modules/lang-painless/build.gradle

```diff
@@ -49,17 +52,41 @@ task apiJavadoc(type: Javadoc) {
   include '**/org/elasticsearch/painless/api/'
   destinationDir = new File(docsDir, 'apiJavadoc')
 }
+
 task apiJavadocJar(type: Jar) {
   classifier = 'apiJavadoc'
   from apiJavadoc
 }
+
 assemble.dependsOn apiJavadocJar
 
-// Reference documentation for Painless's public API.
-task generatePainlessApi(type: JavaExec) {
-  main = 'org.elasticsearch.painless.PainlessDocGenerator'
-  classpath = sourceSets.test.runtimeClasspath
-  args file('../../docs/painless/painless-api-reference')
+/**********************************************
+ *           Context API Generation           *
+ **********************************************/
+
+sourceSets {
+  doc {
+    java {
```

Comment:
This section is redundant actually. The default convention is for source to live in `src/<<source set name>>/java`.

Reply:
Fixed.